engine/src/engine.ts

```diff
@@ -1682,12 +1695,13 @@ export default class Engine {
 
     // tslint:disable-next-line no-shadowed-variable
     const isPossible = (cost: Reward[], income: Reward[]) => {
-      for (const action of actions) {
-        const actionCost = Reward.parse(action.cost);
+      for (const action of data.acts) {
+        const conversion = freeActionConversions[action.action];
+        const actionCost = Reward.parse(conversion.cost);
         if (Reward.includes(cost, actionCost)) {
           // Remove income & cost of action
           const newCost = Reward.merge(cost, Reward.negative(actionCost));
-          let newIncome = Reward.merge(income, Reward.negative(Reward.parse(action.income)));
+          let newIncome = Reward.merge(income, Reward.negative(Reward.parse(conversion.income)));
```

Comment:
you can still use `action.income` / `action.cost` if it's not removed (as per other comments). Setting the cost in available commands and then using when handling command avoids inconsistencies in general across engine upgrades. Like, if the free action `enums` were numbers and a new one is introduced in the middle, or if free actions enums are strings but renamed.